hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java

```diff
@@ -293,27 +347,95 @@ private boolean notHaveAnyExpirableRecord(PartitionContainer partitionContainer)
         }
     }
 
+    /**
+     * Sets last clean-up time before sorting.
+     *
+     * @param partitionContainers partition containers.
+     */
+    private void updateLastCleanupTimesBeforeSorting(List<PartitionContainer> partitionContainers) {
+        for (PartitionContainer partitionContainer : partitionContainers) {
+            partitionContainer.setLastCleanupTimeCopy(partitionContainer.getLastCleanupTime());
+        }
+    }
+
     private Operation createExpirationOperation(int expirationPercentage, int partitionId) {
         return new ClearExpiredOperation(expirationPercentage)
                 .setNodeEngine(nodeEngine)
                 .setCallerUuid(nodeEngine.getLocalMember().getUuid())
                 .setPartitionId(partitionId)
                 .setValidateTarget(false)
-                .setServiceName(SERVICE_NAME);
+                .setServiceName(SERVICE_NAME)
+                .setOperationResponseHandler(this);
     }
 
+    @Override
+    public void sendResponse(Operation op, Object response) {
+        if (canPrimaryDriveExpiration()) {
+            PartitionContainer partitionContainer = partitionContainers[op.getPartitionId()];
+            doBackupExpiration(partitionContainer);
+        }
+    }
 
-    /**
-     * Sets last clean-up time before sorting.
-     *
-     * @param partitionContainers partition containers.
-     */
-    private void updateLastCleanupTimesBeforeSorting(List<PartitionContainer> partitionContainers) {
-        for (PartitionContainer partitionContainer : partitionContainers) {
-            partitionContainer.setLastCleanupTimeCopy(partitionContainer.getLastCleanupTime());
+    public boolean canPrimaryDriveExpiration() {
+        Version version = clusterService.getClusterVersion();
+        return version.isGreaterOrEqual(V3_9);
+    }
+
+    private void doBackupExpiration(PartitionContainer container) {
+        ConcurrentMap<String, RecordStore> maps = container.getMaps();
+        for (RecordStore recordStore : maps.values()) {
+            sendExpiredKeysToBackups(recordStore, false);
         }
     }
 
+    public void sendExpiredKeysToBackups(RecordStore recordStore, boolean checkIfReachedBatch) {
+        InvalidationQueue invalidationQueue = recordStore.getExpiredKeys();
+
+        int size = invalidationQueue.size();
+        if (size == 0 || checkIfReachedBatch && size < MAX_EXPIRED_KEY_COUNT_IN_BATCH) {
```

Comment:
It is already like that. Background expiration task sets `checkIfReachedBatch` to `false` and it scans queue periodically regardless of the batch size.